addons/web/static/tests/views/kanban/kanban_view_tests.js

```diff
@@ -14224,4 +14224,39 @@ QUnit.module("Views", (hooks) => {
         assert.containsOnce(target, ".o_kanban_record:not(.o_kanban_ghost)");
         assert.containsNone(target, ".my_kanban_compiler");
     });
+
+    QUnit.test("Kanban records evaluates the readonly expression", async (assert) => {
+        await makeView({
+            type: "kanban",
+            resModel: "partner",
+            serverData,
+            arch: `<kanban>
+                <templates>
+                    <t t-name="kanban-box">
+                        <div>
+                            <field name="product_id" readonly="4 > 1" />
```

Comment:
<field name="product_id" readonly="context.get('abc')" />
with context: { abc: true }
-> should not allow d&d

<field name="product_id" readonly="display_name == 'coucou'" />
-> should allow d&d